packages/realm-network-transport/src/NetworkTransport.ts

```diff
@@ -36,38 +38,15 @@ export class DefaultNetworkTransport implements NetworkTransport {
     constructor() {
         // Determine the fetch implementation
         if (!DefaultNetworkTransport.fetch) {
-            // Try to get it from the global
-            if (
-                typeof window === "object" &&
-                typeof window.fetch === "function"
-            ) {
-                DefaultNetworkTransport.fetch = window.fetch.bind(window);
-            } else if (isNodeProcess && typeof require === "function") {
-                // Making it harder for the static analyzers see this require call
-                const nodeRequire = require;
-                DefaultNetworkTransport.fetch = nodeRequire("node-fetch");
-            } else {
-                throw new Error(
-                    "DefaultNetworkTransport.fetch must be set before it's used",
-                );
-            }
+            throw new Error(
+                "DefaultNetworkTransport.fetch must be set before it's used",
+            );
         }
         // Determine the AbortController implementation
         if (!DefaultNetworkTransport.AbortController) {
-            if (typeof window === "object" && window.AbortController) {
-                DefaultNetworkTransport.AbortController =
-                    window.AbortController;
-            } else if (isNodeProcess && typeof require === "function") {
-                // Making it harder for the static analyzers see this require call
-                const nodeRequire = require;
-                DefaultNetworkTransport.AbortController = nodeRequire(
-                    "abort-controller",
-                );
-            } else {
-                throw new Error(
-                    "DefaultNetworkTransport.AbortController must be set before it's used",
-                );
-            }
+            throw new Error(
```

Comment:
We could create an `InternalError` which extends `Error` but in my experience, it's not really that common.
I did it with `MongoDBRealmError` because I want to propagate the statusCode, etc., but for an error thrown at runtime if the bundle is malformed, I wouldn't spend the time dealing with this - since no users will ever see this error 🤞

Reply:
I was half asking so I knew what to do for a similar situation in my code. I guess I'll just throw an Error as well.